baremetal/metal3machine_manager.go

```diff
@@ -1259,73 +1286,152 @@ func (m *MachineManager) GetProviderIDAndBMHID() (string, *string) {
 // ClientGetter prototype.
 type ClientGetter func(ctx context.Context, c client.Client, cluster *clusterv1.Cluster) (clientcorev1.CoreV1Interface, error)
 
-// SetNodeProviderID sets the metal3 provider ID on the kubernetes node.
-func (m *MachineManager) SetNodeProviderID(ctx context.Context, providerIDOnM3M *string, clientFactory ClientGetter) error {
-	corev1Remote, err := clientFactory(ctx, m.client, m.Cluster)
+func (m *MachineManager) setNodeProviderID(ctx context.Context, client clientcorev1.CoreV1Interface, node corev1.Node, providerID string) error {
+	oldData, err := json.Marshal(node)
 	if err != nil {
-		return errors.Wrap(err, "Error creating a remote client")
+		return fmt.Errorf("failed to json.Marshal node: %w", err)
+	}
+
+	node.Spec.ProviderID = providerID
+
+	nodeVar := node.DeepCopy()
+	newData, err := json.Marshal(*nodeVar)
+	if err != nil {
+		return fmt.Errorf("failed to json.Marshal node: %w", err)
 	}
+
+	patchBytes, err := strategicpatch.CreateTwoWayMergePatch(oldData, newData, corev1.Node{})
+	if err != nil {
+		return fmt.Errorf("failed to create patch for node %q: %w", node.GetName(), err)
+	}
+	_, err = client.Nodes().Patch(ctx, nodeVar.Name, types.StrategicMergePatchType, patchBytes, metav1.PatchOptions{})
+	if err != nil {
+		return errors.Wrap(err, "unable to update the target node with providerID")
+	}
+
+	return nil
+}
+
+func (m *MachineManager) getMetal3MachineHostname() string {
+	for _, address := range m.Metal3Machine.Status.Addresses {
+		if address.Type == clusterv1.MachineHostName {
+			return address.Address
+		}
+	}
+
+	return ""
+}
+
+// SetProviderID sets the metal3 provider ID on the Metal3Machine.
+func (m *MachineManager) SetProviderID(providerID string) {
+	m.Log.Info("ProviderID set on the Metal3Machine", "providerID", providerID)
+	m.Metal3Machine.Spec.ProviderID = &providerID
+}
+
+func (m *MachineManager) SetDefaultProviderID() {
+	namespace := m.Metal3Machine.GetNamespace()
+	m3mName := m.Metal3Machine.GetName()
+
+	// Ignoring the error because we already checked for the presence of the annotation
```

Comment:
Better to check it also here. This could be called from other places in the future that may not have checked the annotation before.

Reply:
So the comment is outdated now?

Reply:
Yes, removed